src/test/java/com/hotels/jasvorno/JasvornoConverterTest.java

```diff
@@ -109,6 +121,150 @@ public void unknownWithOptional() throws Exception {
+    JasvornoConverter.convertToAvro(datum, schema, UndeclaredFieldBehaviour.IGNORE);
+  }
+
+  @Test
+  public void nullDatumValue() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder.builder().record("record").fields().requiredLong("id").endRecord();
+    Object obj = JasvornoConverter.convertToAvro(null, schema, UndeclaredFieldBehaviour.IGNORE);
+    assertTrue(obj == null);
+  }
+
+  @Test
+  public void nullDatumValueWithModel() throws Exception {
+    Schema schema = SchemaBuilder
+        .builder()
+        .record("record")
+        .fields()
+        .requiredLong("id")
+        .requiredString("name")
+        .endRecord();
+    Object obj = JasvornoConverter.convertToAvro(model, null, schema, UndeclaredFieldBehaviour.IGNORE);
+    assertTrue(obj == null);
+  }
+
+  @Test(expected = JasvornoConverterException.class)
+  public void datumWithNullValue() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder.builder().record("record").fields().requiredString("name").endRecord();
+    ObjectMapper mapper = new ObjectMapper();
+    ObjectNode dataTable = mapper.createObjectNode();
+    JsonNode datum = dataTable.putNull("name");
+    JasvornoConverter.convertToAvro(model, datum, schema, UndeclaredFieldBehaviour.IGNORE);
+  }
+
+  @Test
+  public void datumWithNoValue() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder
+        .record("record")
+        .fields()
+        .name("name")
+        .type()
+        .stringType()
+        .stringDefault("x")
+        .endRecord();
+    JsonNode datum = mapper.readTree("{}");
+    Object obj = JasvornoConverter.convertToAvro(model, datum, schema, UndeclaredFieldBehaviour.IGNORE);
+    assertTrue(obj instanceof IndexedRecord);
+    IndexedRecord record = (IndexedRecord) obj;
+    assertTrue(record.get(0).toString().equals("x"));
+  }
+
+  @Test
+  public void schemaOfTypeMap() throws Exception {
+    Schema schema = SchemaBuilder.map().values().stringType();
+    String json = "{\"name\": \"bob\", \"surname\": \"blue\" }";
+    JsonNode datum = mapper.readTree(json);
+    Object obj = JasvornoConverter.convertToAvro(datum, schema);
+    assertTrue(obj instanceof Map);
+    @SuppressWarnings("unchecked")
+    Map<String, Object> map = (Map<String, Object>) obj;
+    assertTrue(map.size() == 2);
+    assertTrue(map.containsKey("name"));
+    assertTrue(map.containsKey("surname"));
+    assertTrue(map.containsValue("bob"));
+    assertTrue(map.containsValue("blue"));
+  }
+
+  @Test
+  public void schemaOfTypeArray() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder.array().items().intType();
+    JsonNode datum = mapper.readTree("[1,2]");
+    Object obj = JasvornoConverter.convertToAvro(datum, schema);
+    assertTrue(obj instanceof List);
+    @SuppressWarnings("unchecked")
+    List<Object> list = (List<Object>) obj;
+    assertTrue(list.size() == 2);
```

Comment:
Remember how we did this before - you built an expected list to compare against?